webapp/channels/src/utils/performance_telemetry/element_identification.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+/**
+ * A list mapping IDs or CSS classes to regions of the app. In case of nested regions, these are sorted deepest-first.
+ *
+ * The region names map to values of model.AcceptedLCPRegions on the server.
+ */
+const elementIdentifiers = [
+
+    // Post list
+    ['post__content', 'post'],
+    ['create_post', 'post_textbox'],
+
+    // LHS
+    ['SidebarContainer', 'channel_sidebar'],
+    ['team-sidebar', 'team_sidebar'],
+
+    // Header
+    ['channel-header', 'channel_header'],
+    ['global-header', 'global_header'],
+    ['announcement-bar', 'announcement_bar'],
+
+    // Areas of the app
+    ['channel_view', 'center_channel'],
+    ['modal-content', 'modal_content'],
+] as const satisfies Array<[string, string]>;
+
+export type ElementIdentifier = 'other' | typeof elementIdentifiers[number][1];
+
+export function identifyElementRegion(element: Element): ElementIdentifier {
+    let currentElement: Element | null = element;
+
+    while (currentElement) {
```

Comment:
I was a bit worried about the performance of this loop, but it's surprisingly quick. The web app only goes about 40 elements deep at worst